packages/database/src/migrations/20201123061122-AddConfirmedWeeklyDataReportForPSSS-modifies-data.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+import { codeToId, insertObject, generateId } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+const permissionGroupNameToId = async (db, name) => {
+  const record = await db.runSql(`SELECT id FROM permission_group WHERE name = '${name}'`);
+  return record.rows[0] && record.rows[0].id;
+};
+
+const REPORT_CODE = 'PSSS_Confirmed_Weekly_Report';
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  return insertObject(db, 'report', {
+    id: generateId(),
+    code: REPORT_CODE,
+    config: {
+      fetch: {
+        dataElements: [
+          'PSSS_Confirmed_Sites_Reported',
+          'PSSS_Confirmed_Sites',
+          'PSSS_Confirmed_AFR_Cases',
+          'PSSS_Confirmed_AFR_Alert_Threshold_Crossed',
+          'PSSS_Confirmed_DIA_Cases',
+          'PSSS_Confirmed_DIA_Alert_Threshold_Crossed',
+          'PSSS_Confirmed_ILI_Cases',
+          'PSSS_Confirmed_ILI_Alert_Threshold_Crossed',
+          'PSSS_Confirmed_PF_Cases',
+          'PSSS_Confirmed_PF_Alert_Threshold_Crossed',
+          'PSSS_Confirmed_DLI_Cases',
+          'PSSS_Confirmed_DLI_Alert_Threshold_Crossed',
+        ],
+      },
+      transform: [
+        'keyValueByDataElementName',
+        'convertPeriodToWeek',
+        'aggregateFirstValuePerPeriodPerOrgUnit',
```

Comment:
I went with this `aggregateFirstValuePerPeriodPerOrgUnit` as there really only should be 1 value per dataElement per week. This is part of the business logic of PSSS. It's possible to use a `mostRecent` type aggregation here but those really ought to be proceeded with a `sort` which is a bit tricky to do here as we want to transform the period type, and also is a little inefficient.

Given that it's a case that's not meant to happen, I'm happy with going with `first` here (a bit like when you know there should just be a single matching value in an array you use `array.find()`) 👍

Reply:
Sounds good to me! How about naming it `keepFirstValuePerPeriodPerOrgUnit` or `pickFirstValuePerPeriodPerOrgUnit`

Reply:
I was prefixing the names of all these aliases for `aggregate` transformations with `aggregate` as a bit of a hint to the user that 'hey, this is done using aggregation just so ya know` 😛 

But I do agree it's a bit awkward 👍 

How would you feel about just simply: `firstValuePerPeriodPerOrgUnit`? Then we could also have `sumValuesPerPeriod`, or `minValuePerOrgUnit`, etc. ? Or should it really have a verb in front for it to read like a transformation..?

Reply:
I really like `firstValuePerPeriodPerOrgUnit`, reads very natural!